catalyst/utils/components.py

```diff
@@ -19,14 +26,69 @@
 from catalyst.utils.torch import get_device
 
 
+def _patch_forward(model):
+    import apex
+
+    input_caster_lambda = (
+        lambda tensor: tensor.to(
+            apex.amp._amp_state.opt_properties.options["cast_model_type"]
+        )
+        if tensor.is_floating_point()
+        else tensor
+    )
+    output_caster_lambda = (
+        lambda tensor: tensor.to(
+            apex.amp._amp_state.opt_properties.options.get(
+                "cast_model_outputs", torch.float32
+            )
+        )
+        if tensor.is_floating_point()
+        else tensor
+    )
+
+    model.forward = lambda *args, old_fwd=model.forward, input_caster=input_caster_lambda, output_caster=output_caster_lambda, **kwargs: apex.amp._initialize.applier(
```

Comment:
is it possible to use ``**kwargs`` with this hack?

Reply:
yep